extensions/sql/asset/index/src/test/java/org/eclipse/dataspaceconnector/sql/asset/index/SqlAssetIndexTest.java

```diff
@@ -0,0 +1,260 @@
+        sqlAssetIndex.accept(asset, getDataAddress());
+
+        assertThatThrownBy(() -> sqlAssetIndex.accept(asset, getDataAddress()))
+                .isInstanceOf(EdcPersistenceException.class)
+                .hasMessageContaining(String.format("Cannot persist. Asset with ID '%s' already exists.", asset.getId()));
+    }
+
+    @Test
+    @DisplayName("Accept an asset entry that doesn't exist yet")
+    void acceptAssetEntry_doesNotExist() {
+        var assetExpected = getAsset("id1");
+        sqlAssetIndex.accept(new AssetEntry(assetExpected, getDataAddress()));
+
+        var assetFound = sqlAssetIndex.findById("id1");
+
+        assertThat(assetFound).isNotNull();
+        assertThat(assetFound.getId()).isEqualTo(assetExpected.getId());
+    }
+
+    @Test
+    @DisplayName("Accept an asset entry that already exists")
+    void acceptEntry_exists() {
+        var asset = getAsset("id1");
+        sqlAssetIndex.accept(new AssetEntry(asset, getDataAddress()));
+
+        assertThatThrownBy(() -> sqlAssetIndex.accept(asset, getDataAddress()))
+                .isInstanceOf(EdcPersistenceException.class)
+                .hasMessageContaining(String.format("Cannot persist. Asset with ID '%s' already exists.", asset.getId()));
+    }
+
+    @Test
+    @DisplayName("Delete an asset that doesn't exist")
+    void deleteAsset_doesNotExist() {
+        var assetDeleted = sqlAssetIndex.deleteById("id1");
+
+        assertThat(assetDeleted).isNull();
+    }
+
+    @Test
+    @DisplayName("Delete an asset that exists")
+    void deleteAsset_exists() {
+        var asset = getAsset("id1");
+        sqlAssetIndex.accept(asset, getDataAddress());
+
+        var assetDeleted = sqlAssetIndex.deleteById("id1");
+
+        assertThat(assetDeleted).isNotNull();
+        assertThat(assetDeleted.getId()).isEqualTo("id1");
+    }
+
+    @Test
+    @DisplayName("Query assets with selector expression")
+    void queryAsset_selectorExpression() {
+        var asset1 = getAsset("id1");
+        sqlAssetIndex.accept(asset1, getDataAddress());
+        var asset2 = getAsset("id2");
+        sqlAssetIndex.accept(asset2, getDataAddress());
+
+        var assetsFound = sqlAssetIndex.queryAssets(AssetSelectorExpression.Builder.newInstance().build());
+
+        assertThat(assetsFound).isNotNull();
+        assertThat(assetsFound.count()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("Query assets with query spec")
+    void queryAsset_querySpec() {
+        for (var i = 1; i <= 10; i++) {
+            var asset = getAsset("id" + i);
+            sqlAssetIndex.accept(asset, getDataAddress());
+        }
+
+        var assetsFound = sqlAssetIndex.queryAssets(getQuerySpec());
+
+        assertThat(assetsFound).isNotNull();
+        assertThat(assetsFound.count()).isEqualTo(3);
+    }
+
+    @Test
+    @DisplayName("Query assets with query spec and short asset count")
+    void queryAsset_querySpecShortCount() {
+        for (var i = 1; i <= 4; i++) {
```

Comment:
nit: use `IntStream.range`, because that will give you an elegant way to collect the created objects into a list and use them for assertions later.